server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -98,6 +98,11 @@ public final class IndexModule {
 
     public static final Setting<Boolean> NODE_STORE_ALLOW_MMAP = Setting.boolSetting("node.store.allow_mmap", true, Property.NodeScope);
 
+    /**
+     * Setting to enable or disable all fsync operations on a node.
+     */
+    public static final Setting<Boolean> NODE_STORE_USE_FSYNC = Setting.boolSetting("node.store.use_fsync", true, Property.NodeScope);
```

Comment:
A little unfortunate how long this PR became when doing this via a setting. Could do the same thing in 30 lines using a system property probably but I think the test randomisation and general alignment with how we do these things makes it worth to go through the noise of adding a setting here.